### ISSUE TITLE
Reporting an empty dict does not work

### ISSUE THREAD

**Issue by Mano3** [allegroai/trains#265]
Hi, it seems the following does not work:
`task.connect({'a':{},'b':'moshe'}, name='wat')`
It seems a is absent from the recorded arguments while b is there.
Thanks.

**Comment by bmartinn** [comment]
Hi @Mano3 
This is "kind-of" on purpose, the thing is, nested dictionaries are flattened in the "connect" function, making sure we can store the nested dictionaries in a single key/value list (e.g. `{'a_dict':{'a': 1, 'a': 2}}` becomes `[('a_dict/a', 1), ('a_dict/b', 2),]` ). Since we have an empty dictionary, there is nothing to store.

That said, we can store it as empty string '', or we can store it as '{}' , I think the main issue is how would one use it?
On top of it, in theory you could have both `'a' = {'nested': 1}` and `'a/nested' = 1` ...
What is the use case here? I mean how would a user know how to fill in the dictionary keys ?

WDYT?

**Comment by Mano3** [comment]
The use case is optional key values for matplotlib plot function. In most cases, the dictionary is empty as we don't need need to change anything. But when we do, we want these settings store in a dictionary format, as the matplotlib parameters are only a fraction of a larger configuration file which includes it.

I've got a suggestion.
Maybe instead of flattening as you do, indent the arguments related, when detecting a dictionary. In that case, when the dictionary is empty, the indented rows would have only one empty row.

Hard to visualize through text, but I hope you can understand.

**Comment by bmartinn** [comment]
Hi @Mano3 
I think I see what you mean, and I might have an idea.
Regrading the empty dict, if we store `{}` , we can always later parse this string as json string, allowing you to create an arbitrary dict in remote execution. That said, I'm not a big fan of editing json inside a UI (it is so easy to make a mistake, only to find it when your are actually running the experiment).

Specifically to your case, how about using `task.connect_configuration`, you can see an example in the [documentation](https://allegro.ai/docs/examples/reporting/model_config/#configuration-dictionary)
Basically it allows you to fully edit any complex dictionary, directly from the UI, in a json alike syntax that is very forgiving (it's actually called HOCON)

WDYT?

**Comment by patrickdasilva-clearml** [comment]
Hi @Mano3 , we're closing this issue as stale since it's been created too long ago. If you think the issue is still relevant, please make sure the issue persists on a recent version of ClearML (ideally the latest) before re-opening the issue. We'll gladly look at it again.
